Codegen fails with `CompileException`: Cannot determine simple type name "com"

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/fury/issues) and found no similar issues.


### Version

Fury 0.10.0
OpenJDK Temurin-21.0.5+11

### Component(s)

Java

### Minimal reproduce step

```scala
package com.example.reproduce

import org.apache.fury.Fury
import org.apache.fury.format.encoder.{Encoders, RowEncoder}

object ReproduceFuryRow {
  private val fury = Fury
    .builder()
    .requireClassRegistration(false)
    .build()

  case class InnerClass(
      a: Int
  )

  case class MainClass(
      arr: Array[InnerClass]
//      z: InnerClass = null
  )

  def main(args: Array[String]): Unit = {
    val encoder: RowEncoder[MainClass] = Encoders.bean(classOf[MainClass], fury)
    val obj = MainClass(Array(InnerClass(1), InnerClass(2)))
    val row = encoder.toRow(obj)

    val sp = row.getArray(0)
    val inner = sp.getStruct(0)
    println(s"Value = ${inner.getInt32(0)}")
  }
}
```

### What did you expect to see?

I expected it to run successfully, like it does when we uncomment the `z: InnerClass = null` line.

### What did you see instead?

```
[error] Exception in thread "main" org.apache.fury.format.encoder.EncoderException: Create encoder failed,
[error] beanClass: class com.example.reproduce.ReproduceFuryRow$MainClass
[error] 	at org.apache.fury.format.encoder.Encoders.bean(Encoders.java:217)
[error] 	at org.apache.fury.format.encoder.Encoders.bean(Encoders.java:75)
[error] 	at org.apache.fury.format.encoder.Encoders.bean(Encoders.java:69)
[error] 	at com.example.reproduce.ReproduceFuryRow$.main(ReproduceFuryRow.scala:25)
[error] 	at com.example.reproduce.ReproduceFuryRow.main(ReproduceFuryRow.scala)
[error] Caused by: org.apache.fury.codegen.CodegenException: Compile error:
[error] com.example.reproduce.ReproduceFuryRow_MainClassRowCodec205029188_243575009:
[error] /* 0001 */ package com.example.reproduce;
[error] /* 0002 */
[error] /* 0003 */ import org.apache.fury.format.row.binary.*;
[error] /* 0004 */ import org.apache.fury.format.row.binary.writer.*;
[error] /* 0005 */ import org.apache.arrow.vector.types.pojo.*;
[error] /* 0006 */ import org.apache.arrow.vector.types.pojo.Field;
[error] /* 0007 */ import org.apache.arrow.vector.types.pojo.Schema;
[error] /* 0008 */ import org.apache.fury.format.row.Row;
[error] /* 0009 */ import org.apache.fury.format.row.ArrayData;
[error] /* 0010 */ import org.apache.fury.format.row.MapData;
[error] /* 0011 */ import org.apache.fury.format.row.binary.BinaryRow;
[error] /* 0012 */ import org.apache.fury.format.row.binary.BinaryArray;
[error] /* 0013 */ import org.apache.fury.format.row.binary.BinaryMap;
[error] /* 0014 */
[error] /* 0015 */ public final class ReproduceFuryRow_MainClassRowCodec205029188_243575009 implements org.apache.fury.format.encoder.GeneratedRowEncoder {
[error] /* 0016 */
[error] /* 0017 */   private final Class beanClass;
[error] /* 0018 */   private Schema schema;
[error] /* 0019 */   private BinaryRowWriter rowWriter;
[error] /* 0020 */   private org.apache.fury.Fury fury;
[error] /* 0021 */   private final BinaryArrayWriter arrayWriter;
[error] /* 0022 */   private final BinaryRowWriter innerClassRowWriter;
[error] /* 0023 */   private final com.example.reproduce.ReproduceFuryRow_InnerClassRowCodec205029188_1943855334 reproduceFuryRow_InnerClassRowCodec205029188_1943855334;
[error] /* 0024 */
[error] /* 0025 */   public ReproduceFuryRow_MainClassRowCodec205029188_243575009(java.lang.Object[] references) {
[error] /* 0026 */       schema = (Schema)references[0];
[error] /* 0027 */       rowWriter = (BinaryRowWriter)references[1];
[error] /* 0028 */       fury = (org.apache.fury.Fury)references[2];
[error] /* 0029 */
[error] /* 0030 */       beanClass = com.example.reproduce.ReproduceFuryRow$MainClass.class;
[error] /* 0031 */       Field field1 = org.apache.fury.format.type.DataTypes.fieldOfSchema(schema, 0);
[error] /* 0032 */       BinaryArrayWriter binaryArrayWriter = new BinaryArrayWriter(field1, rowWriter);
[error] /* 0033 */       arrayWriter = binaryArrayWriter;
[error] /* 0034 */       Field elemField2 = org.apache.fury.format.type.DataTypes.arrayElementField(field1);
[error] /* 0035 */       Schema schema3 = org.apache.fury.format.type.DataTypes.schemaFromStructField(elemField2);
[error] /* 0036 */       BinaryRowWriter binaryRowWriter = new BinaryRowWriter(schema3, arrayWriter);
[error] /* 0037 */       innerClassRowWriter = binaryRowWriter;
[error] /* 0038 */       java.lang.Object[] arr0 = new java.lang.Object[] {schema3, binaryRowWriter, fury};
[error] /* 0039 */       com.example.reproduce.ReproduceFuryRow_InnerClassRowCodec205029188_1943855334 generatedRowEncoder = new com.example.reproduce.ReproduceFuryRow_InnerClassRowCodec205029188_1943855334(arr0);
[error] /* 0040 */       reproduceFuryRow_InnerClassRowCodec205029188_1943855334 = generatedRowEncoder;
[error] /* 0041 */   }
[error] /* 0042 */
[error] /* 0043 */   @Override public final BinaryRow toRow(Object obj) {
[error] /* 0044 */       com.example.reproduce.ReproduceFuryRow.MainClass mainClass1 = (com.example.reproduce.ReproduceFuryRow.MainClass)obj;
[error] /* 0045 */       Object object1 = org.apache.fury.memory.Platform.getObject(mainClass1, 12L);
[error] /* 0046 */       com.example.reproduce.ReproduceFuryRow.InnerClass[] arr = (com.example.reproduce.ReproduceFuryRow.InnerClass[])object1;
[error] /* 0047 */       if ((arr == null)) {
[error] /* 0048 */           rowWriter.setNullAt(0);
[error] /* 0049 */       } else {
[error] /* 0050 */           int writerIndex = rowWriter.writerIndex();
[error] /* 0051 */           int length = arr.length;
[error] /* 0052 */           arrayWriter.reset(length);
[error] /* 0053 */           int len = arr.length;
[error] /* 0054 */           int i = 0;
[error] /* 0055 */           while (i < len) {
[error] /* 0056 */               com.example.reproduce.ReproduceFuryRow.InnerClass elemValue = arr[i];
[error] /* 0057 */               if ((elemValue == null)) {
[error] /* 0058 */                   arrayWriter.setNullAt(i);
[error] /* 0059 */               } else {
[error] /* 0060 */                   int writerIndex1 = arrayWriter.writerIndex();
[error] /* 0061 */                   innerClassRowWriter.reset();
[error] /* 0062 */                   reproduceFuryRow_InnerClassRowCodec205029188_1943855334.toRow(elemValue);
[error] /* 0063 */                   int writerIndex2 = arrayWriter.writerIndex();
[error] /* 0064 */                   arrayWriter.setOffsetAndSize(i, writerIndex1, (writerIndex2 - writerIndex1));
[error] /* 0065 */               }
[error] /* 0066 */               i++;
[error] /* 0067 */           }
[error] /* 0068 */           int writerIndex3 = rowWriter.writerIndex();
[error] /* 0069 */           rowWriter.setOffsetAndSize(0, writerIndex, (writerIndex3 - writerIndex));
[error] /* 0070 */       }
[error] /* 0071 */       BinaryRow binaryRow = rowWriter.getRow();
[error] /* 0072 */       return binaryRow;
[error] /* 0073 */   }
[error] /* 0074 */
[error] /* 0075 */   @Override public final Object fromRow(BinaryRow row) {
[error] /* 0076 */       Object instance = org.apache.fury.memory.Platform.newInstance(com.example.reproduce.ReproduceFuryRow$MainClass.class);
[error] /* 0077 */       com.example.reproduce.ReproduceFuryRow.MainClass mainClass2 = (com.example.reproduce.ReproduceFuryRow.MainClass)instance;
[error] /* 0078 */       boolean value = row.isNullAt(0);
[error] /* 0079 */       if ((!value)) {
[error] /* 0080 */           BinaryArray binaryArray1 = row.getArray(0);
[error] /* 0081 */           int value1 = binaryArray1.numElements();
[error] /* 0082 */           com.example.reproduce.ReproduceFuryRow.InnerClass[] arr1 = new com.example.reproduce.ReproduceFuryRow.InnerClass[value1];
[error] /* 0083 */           int len1 = binaryArray1.numElements();
[error] /* 0084 */           int i1 = 0;
[error] /* 0085 */           while (i1 < len1) {
[error] /* 0086 */               if (!binaryArray1.isNullAt(i1)) {
[error] /* 0087 */                   BinaryRow elemValue1 = binaryArray1.getStruct(i1);
[error] /* 0088 */                   Object object2 = reproduceFuryRow_InnerClassRowCodec205029188_1943855334.fromRow(elemValue1);
[error] /* 0089 */                   com.example.reproduce.ReproduceFuryRow.InnerClass bean = (com.example.reproduce.ReproduceFuryRow.InnerClass)object2;
[error] /* 0090 */                   arr1[i1] = bean;
[error] /* 0091 */               }
[error] /* 0092 */               i1++;
[error] /* 0093 */           }
[error] /* 0094 */           org.apache.fury.memory.Platform.putObject(mainClass2, 12L, arr1);
[error] /* 0095 */       }
[error] /* 0096 */       return mainClass2;
[error] /* 0097 */   }
[error] /* 0098 */
[error] /* 0099 */ }
[error] 	at org.apache.fury.codegen.JaninoUtils.toBytecode(JaninoUtils.java:133)
[error] 	at org.apache.fury.codegen.JaninoUtils.toBytecode(JaninoUtils.java:73)
[error] 	at org.apache.fury.codegen.CodeGenerator.compile(CodeGenerator.java:145)
[error] 	at org.apache.fury.codegen.CodeGenerator.compile(CodeGenerator.java:117)
[error] 	at org.apache.fury.format.encoder.Encoders.loadCls(Encoders.java:701)
[error] 	at org.apache.fury.format.encoder.Encoders.loadOrGenRowCodecClass(Encoders.java:648)
[error] 	at org.apache.fury.format.encoder.Encoders.bean(Encoders.java:136)
[error] 	... 4 more
[error] Caused by: org.apache.fury.shaded.org.codehaus.commons.compiler.CompileException: File 'com/example/reproduce/ReproduceFuryRow_MainClassRowCodec205029188_243575009.java', Line 23, Column 11: File 'com/example/reproduce/ReproduceFuryRow_MainClassRowCodec205029188_243575009.java', Line 23, Column 11: Cannot determine simple type name "com"
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getReferenceType(UnitCompiler.java:7007)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getReferenceType(UnitCompiler.java:6886)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getReferenceType(UnitCompiler.java:6899)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getReferenceType(UnitCompiler.java:6899)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getReferenceType(UnitCompiler.java:6899)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getType2(UnitCompiler.java:6857)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$14800(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$24.visitReferenceType(UnitCompiler.java:6755)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$24.visitReferenceType(UnitCompiler.java:6752)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$ReferenceType.accept(Java.java:4289)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getType(UnitCompiler.java:6752)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getRawType(UnitCompiler.java:6748)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$1300(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$30.getType(UnitCompiler.java:8645)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getType2(UnitCompiler.java:7241)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$18300(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$26.visitFieldAccess(UnitCompiler.java:6818)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$26.visitFieldAccess(UnitCompiler.java:6815)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$FieldAccess.accept(Java.java:4697)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getType(UnitCompiler.java:6815)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$15100(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$25.visitLvalue(UnitCompiler.java:6776)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$25.visitLvalue(UnitCompiler.java:6773)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$Lvalue.accept(Java.java:4528)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getType(UnitCompiler.java:6773)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$14400(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$23.visitRvalue(UnitCompiler.java:6721)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$23.visitRvalue(UnitCompiler.java:6712)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$Rvalue.accept(Java.java:4495)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getType(UnitCompiler.java:6712)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getType2(UnitCompiler.java:7225)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$18100(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$26.visitAmbiguousName(UnitCompiler.java:6816)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$26.visitAmbiguousName(UnitCompiler.java:6815)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$AmbiguousName.accept(Java.java:4603)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getType(UnitCompiler.java:6815)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$15100(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$25.visitLvalue(UnitCompiler.java:6776)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$25.visitLvalue(UnitCompiler.java:6773)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$Lvalue.accept(Java.java:4528)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getType(UnitCompiler.java:6773)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$14400(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$23.visitRvalue(UnitCompiler.java:6721)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$23.visitRvalue(UnitCompiler.java:6712)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$Rvalue.accept(Java.java:4495)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getType(UnitCompiler.java:6712)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.findIMethod(UnitCompiler.java:9423)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compileGet2(UnitCompiler.java:5263)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$9300(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$16.visitMethodInvocation(UnitCompiler.java:4766)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$16.visitMethodInvocation(UnitCompiler.java:4742)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$MethodInvocation.accept(Java.java:5470)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compileGet(UnitCompiler.java:4742)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compileGetValue(UnitCompiler.java:5885)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:4121)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$6100(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$13.visitMethodInvocation(UnitCompiler.java:4096)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$13.visitMethodInvocation(UnitCompiler.java:4071)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$MethodInvocation.accept(Java.java:5470)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:4071)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:2524)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$1800(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitExpressionStatement(UnitCompiler.java:1581)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitExpressionStatement(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$ExpressionStatement.accept(Java.java:3209)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compileStatements(UnitCompiler.java:1662)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:1647)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$1700(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitBlock(UnitCompiler.java:1580)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitBlock(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$Block.accept(Java.java:3115)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:2638)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$1900(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitIfStatement(UnitCompiler.java:1582)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitIfStatement(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$IfStatement.accept(Java.java:3284)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compileStatements(UnitCompiler.java:1662)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:1647)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$1700(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitBlock(UnitCompiler.java:1580)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitBlock(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$Block.accept(Java.java:3115)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:2002)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$2200(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitWhileStatement(UnitCompiler.java:1585)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitWhileStatement(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$WhileStatement.accept(Java.java:3389)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compileStatements(UnitCompiler.java:1662)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:1647)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$1700(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitBlock(UnitCompiler.java:1580)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitBlock(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$Block.accept(Java.java:3115)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:2638)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$1900(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitIfStatement(UnitCompiler.java:1582)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$6.visitIfStatement(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$IfStatement.accept(Java.java:3284)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:1576)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compileStatements(UnitCompiler.java:1662)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:3665)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:3330)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compileDeclaredMethods(UnitCompiler.java:1448)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compileDeclaredMethods(UnitCompiler.java:1421)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:830)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:443)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$400(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$3.visitPackageMemberClassDeclaration(UnitCompiler.java:423)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$3.visitPackageMemberClassDeclaration(UnitCompiler.java:419)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$PackageMemberClassDeclaration.accept(Java.java:1688)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:419)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:393)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.access$000(UnitCompiler.java:237)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$2.visitCompilationUnit(UnitCompiler.java:364)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler$2.visitCompilationUnit(UnitCompiler.java:362)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Java$CompilationUnit.accept(Java.java:371)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compileUnit(UnitCompiler.java:362)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Compiler.compile2(Compiler.java:271)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.Compiler.compile(Compiler.java:213)
[error] 	at org.apache.fury.codegen.JaninoUtils.toBytecode(JaninoUtils.java:115)
[error] 	... 10 more
[error] Caused by: org.apache.fury.shaded.org.codehaus.commons.compiler.CompileException: File 'com/example/reproduce/ReproduceFuryRow_MainClassRowCodec205029188_243575009.java', Line 23, Column 11: Cannot determine simple type name "com"
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.compileError(UnitCompiler.java:13080)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getRawReferenceType(UnitCompiler.java:7175)
[error] 	at org.apache.fury.shaded.org.codehaus.janino.UnitCompiler.getReferenceType(UnitCompiler.java:7005)
[error] 	... 135 more
```

### Anything Else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/java/fury-core/src/main/java/org/apache/fury/type/TypeUtils.java
+++ b/java/fury-core/src/main/java/org/apache/fury/type/TypeUtils.java
@@ -686,21 +686,37 @@ public class TypeUtils {
       typeRefs.addAll(getAllTypeArguments(typeRef));
     }
 
-    typeRefs.stream()
-        .filter(typeToken -> isBean(getRawType(typeToken)))
-        .forEach(
-            typeToken -> {
-              Class<?> cls = getRawType(typeToken);
-              beans.add(cls);
-              if (walkedTypePath.contains(typeToken)) {
-                throw new UnsupportedOperationException(
-                    "cyclic type is not supported. walkedTypePath: " + walkedTypePath);
-              } else {
-                LinkedHashSet<TypeRef<?>> newPath = new LinkedHashSet<>(walkedTypePath);
-                newPath.add(typeToken);
-                beans.addAll(listBeansRecursiveInclusive(cls, newPath));
-              }
-            });
+    for (TypeRef<?> typeToken : typeRefs) {
+      Class<?> type = getRawType(typeToken);
+      if (isBean(type)) {
+        beans.add(type);
+        if (walkedTypePath.contains(typeToken)) {
+          throw new UnsupportedOperationException(
+              "cyclic type is not supported. walkedTypePath: " + walkedTypePath);
+        } else {
+          LinkedHashSet<TypeRef<?>> newPath = new LinkedHashSet<>(walkedTypePath);
+          newPath.add(typeToken);
+          beans.addAll(listBeansRecursiveInclusive(type, newPath));
+        }
+      } else if (isCollection(type)) {
+        TypeRef<?> elementType = getElementType(typeToken);
+        LinkedHashSet<TypeRef<?>> newPath = new LinkedHashSet<>(walkedTypePath);
+        newPath.add(elementType);
+        beans.addAll(listBeansRecursiveInclusive(elementType.getClass(), newPath));
+      } else if (isMap(type)) {
+        Tuple2<TypeRef<?>, TypeRef<?>> mapKeyValueType = getMapKeyValueType(typeToken);
+        LinkedHashSet<TypeRef<?>> newPath = new LinkedHashSet<>(walkedTypePath);
+        newPath.add(mapKeyValueType.f0);
+        newPath.add(mapKeyValueType.f1);
+        beans.addAll(listBeansRecursiveInclusive(mapKeyValueType.f0.getRawType(), newPath));
+        beans.addAll(listBeansRecursiveInclusive(mapKeyValueType.f1.getRawType(), newPath));
+      } else if (type.isArray()) {
+        Class<?> arrayComponent = getArrayComponent(type);
+        LinkedHashSet<TypeRef<?>> newPath = new LinkedHashSet<>(walkedTypePath);
+        newPath.add(TypeRef.of(arrayComponent));
+        beans.addAll(listBeansRecursiveInclusive(arrayComponent, newPath));
+      }
+    }
     return beans;
   }
 

--- a/java/fury-core/src/test/java/org/apache/fury/type/TypeUtilsTest.java
+++ b/java/fury-core/src/test/java/org/apache/fury/type/TypeUtilsTest.java
@@ -132,6 +132,27 @@ public class TypeUtilsTest {
     assertEquals(classes.size(), 2);
   }
 
+  public static class ListBeanGenericFoo1 {}
+
+  public static class ListBeanGenericFoo2 {}
+
+  public static class ListBeanGenericFoo3 {}
+
+  public static class ListBeanGenericFoo4 {}
+
+  public static class ListBeanGeneric {
+    List<ListBeanGenericFoo1> f1;
+    ListBeanGenericFoo2[] f2;
+    Map<ListBeanGenericFoo3, ListBeanGenericFoo4> f3;
+  }
+
+  @Test
+  public void listBeanGeneric() {
+    LinkedHashSet<Class<?>> classes = TypeUtils.listBeansRecursiveInclusive(ListBeanGeneric.class);
+    // System.out.println(classes);
+    assertEquals(classes.size(), 5);
+  }
+
   @Test
   public void isBean() {
     Assert.assertTrue(TypeUtils.isBean(BeanA.class));

